libs/ui-lib/lib/ocm/components/featureSupportLevels/FeatureSupportLevelProvider.tsx

```diff
@@ -1,93 +1,44 @@
 import useInfraEnv from '../../hooks/useInfraEnv';
-import { isFeatureSupportedAndAvailable } from '../newFeatureSupportLevels/newFeatureStateUtils';
-
-export type SupportLevelProviderProps = PropsWithChildren<{
+import {
+  NewFeatureSupportLevelContextProvider,
+  NewFeatureSupportLevelData,
+  NewFeatureSupportLevelMap,
+} from '../../../common/components/newFeatureSupportLevels';
+import useSupportLevelsAPI from '../../hooks/useSupportLevelsAPI';
+import { TFunction } from 'i18next';
+
+export type NewSupportLevelProviderProps = PropsWithChildren<{
   clusterFeatureUsage?: string;
   openshiftVersion?: string;
   loadingUi: React.ReactNode;
   cluster?: Cluster;
+  cpuArchitecture?: string;
 }>;
 
-const getFeatureSupportLevelsMap = (
-  featureSupportLevels: FeatureSupportLevels,
-): FeatureSupportLevelsMap => {
-  try {
-    const featureSupportLevelsMap: FeatureSupportLevelsMap = {};
-    for (const { openshiftVersion, features } of featureSupportLevels) {
-      if (!openshiftVersion || !features) {
-        continue;
-      }
-      featureSupportLevelsMap[openshiftVersion] = {};
-      for (const { featureId, supportLevel } of features) {
-        if (!featureId || !supportLevel) {
-          continue;
-        }
-        featureSupportLevelsMap[openshiftVersion][featureId] = supportLevel;
-      }
-    }
-    return featureSupportLevelsMap;
-  } catch (err) {
-    captureException(err, 'Failed to parse feature support levels', Sentry.Severity.Warning);
-    return {};
-  }
-};
-
-const getVersionSupportLevelsMap = (
-  versionName: string,
-  supportLevelData: FeatureSupportLevelsMap,
-): FeatureIdToSupportLevel | undefined => {
-  // There is only one feature-support-level item per minor OpenShift version
-  // To solve the problem of partial matching (4.1 vs 4.11), we sort the list from most recent to least recent.
-  const versionKey = Object.keys(supportLevelData)
-    .reverse()
-    .find((key) => versionName.startsWith(key));
-  if (!versionKey) {
-    return undefined;
-  }
-
-  return supportLevelData[versionKey];
-};
-
-export const getFeatureSupported = (
-  openshiftVersion: string,
-  featureSupportLevels: FeatureSupportLevels,
-  featureId: FeatureId,
-) => {
-  const featureSupportLevelsMap = getFeatureSupportLevelsMap(featureSupportLevels);
-  const versionSupportLevels = getVersionSupportLevelsMap(
-    openshiftVersion,
-    featureSupportLevelsMap,
-  );
-  return versionSupportLevels && isFeatureSupportedAndAvailable(versionSupportLevels[featureId]);
+export const getFeatureSupported = (featureSupportLevels: SupportLevels, featureId: FeatureId) => {
+  return featureSupportLevels && isFeatureSupportedAndAvailable(featureSupportLevels[featureId]);
 };
 
-export const FeatureSupportLevelProvider: React.FC<SupportLevelProviderProps> = ({
+export const NewFeatureSupportLevelProvider: React.FC<NewSupportLevelProviderProps> = ({
   cluster,
   children,
   loadingUi,
+  cpuArchitecture,
+  openshiftVersion,
```

Comment:
`openshiftVersion` is always passed from the `cluster` object, which you also have in the props. So no need for the extra prop.

Reply:
In NewClusterPage we don't have the cluster. Then we need to pass the openshiftVersion and cpuArchitecture to the provider.